### PR TITLE
Add dive bombing capability to CAS supports

### PR DESCRIPTION
**Pull request by jaj22**
### What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [X] Enhancement

### What have you changed and why?
Added a dive bombing routine to CAS supports so that WW2 factions can kill tanks. We don't actually have any WW2 factions yet so it's a bit theoretical, but I tested IFA aircraft vs IFA tanks and the success rate is pretty high. Towards 100% with larger tanks (or the PE-2), 70-80% against smaller tanks with the Ju-87. Moving targets aren't a problem although they could dodge after the bomb release.

It's a fairly limited implementation. CAS aircraft with the bombRacks variable specified in loadouts will do a dive bombing run first, dropping all their bombs at once. After that, they'll switch to the standard CAS routine but as they only have guns, they'll quit after one pass. Their strafing firepower is quite limited and they're not hard to hit with AAA so this is probably fine.

Ju-87s will occasionally kill their pilot with the bomb at the current settings. This is fine.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [X] Yes (Please provide further detail below.)

Well, we don't actually have a WW2 faction yet, so I suspect this isn't going to be live-tested for a while.

### How can the changes be tested?
- Load IFA.
- Set CAS plane in vanilla AAF template to either `LIB_Ju87` or `LIB_Pe2`.
- Start game with vanilla AAF airports somewhere near.
- Place a tank with zeus.
- With the tank selected in zeus, run this:
```
onEachFrame {
 (allCurators#0) addCuratorEditableObjects [allUnits];
 (allCurators#0) addCuratorEditableObjects [vehicles, true];
};
logLevel = 3;
_targ = curatorSelected#0#0;
["CAS", Occupants, "defence", 500, _targ, getPosATL _targ, 1, 0] call A3A_fnc_createSupport;
```
Test code trash left in. Remove if unnecessary.

### CODE REVIEW

File: A3A/addons/core/functions/Supports/fn_SUP_CASApproach.sqf [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    params ["_plane", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile", "_gunner"];
+
+    //Reduce the available ammo for internal logic
+    private _ammoHM = _plane getVariable "ammoCount";
+    private _ammoCount = _ammoHM get _weapon;
+    if (!isNil "_ammoCount") then { _ammoHM set [_weapon, _ammoCount-1] };
+
+    // Everything else requires an existing target
+    private _targetObj = _plane getVariable ["currentTarget", objNull];
+    if (isNull _targetObj) exitWith {};
```
[the remark this answers is not in the document]
Comment (jaj22) [reply A3A/addons/core/functions/Supports/fn_SUP_CASApproach.sqf:66]: No, it's an acceptable case and could be extremely spammy. Target can be garaged, or die and be garbage cleaned before the shot was fired.

File: A3A/addons/core/functions/Supports/fn_SUP_CASApproach.sqf [hunk 2]
```diff
@@ -0,0 +1,321 @@
+
```
Comment (killerswin2) [inline A3A/addons/core/functions/Supports/fn_SUP_CASApproach.sqf:1]: Header file header please.
Just make a simple one

**Review by killerswin2 — APPROVED** [review]
I'm not seeing anything terrible in here. Just need to add some headers but other than that, this is good for merging to unstable.

**Comment by jaj22** [comment]
I'll do the headers before the merge, forgot.

**Review by Bob-Murphy — APPROVED** [review]
lgtm

**Comment by Bob-Murphy** [comment]
Set to `Changes requested` based on the missing headers.

**Comment by Bob-Murphy** [comment]
As discussed, please clean up the CUP stuff.
Cheers.

**Comment by killerswin2** [comment]
Fixed. Ready for merge @Bob-Murphy

File: A3A/addons/config_fixes/config.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    class ADDON {
+        name = COMPONENT_NAME;
+        units[] = {};
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        #if __has_include("\CUP\Weapons\CUP_Weapons_DynamicLoadout\mk82\CUP_mk82_pod.p3d")
+        requiredAddons[] = {"A3_Weapons_F", "CUP_Weapons_DynamicLoadout"};
+        #else
+        requiredAddons[] = {"A3_Weapons_F"};
+        #endif
```
Comment (jaj22) [inline A3A/addons/config_fixes/config.cpp:13]: Oh, for load order. Ugh :P

Comment (killerswin2) [reply A3A/addons/config_fixes/config.cpp:13]: yup

**Pull request merged**